ste/sender-blockBlob.go

```diff
@@ -62,6 +64,29 @@ func newBlockBlobSenderBase(jptm IJobPartTransferMgr, destination string, p pipe
 	chunkSize := transferInfo.BlockSize
 	srcSize := transferInfo.SourceSize
 	numChunks := getNumChunks(srcSize, chunkSize)
+	toGiB := func(bytes int64) float64 {
+		return float64(bytes) / float64(1024*1024*1024)
+	}
+	if chunkSize > common.MaxBlockBlobBlockSize {
+		// mercy, please
+		return nil, fmt.Errorf("Source size %d exceeds maxmimum allowed limit for a BlockBlob", srcSize)
+	}
+
+	if chunkSize >= jptm.CacheLimiter().Limit() {
+		return nil, fmt.Errorf("Cannot use a block size of %.2fGiB. AzCopy is limited to use only %.2fGiB of memory",
+			toGiB(chunkSize), toGiB(jptm.CacheLimiter().Limit()))
+	}
+
+	if common.AzCopyParallelChunkCountThreshold >= jptm.CacheLimiter().Limit()/chunkSize {
+		glcm := common.GetLifecycleMgr()
+		msg := fmt.Sprintf("Using a blocksize of %.2fGiB for file %s. AzCopy is limited to use %.2fGiB of memory."+
+			"Consider providing atleast %.2fGiB to AzCopy, using environment variable AZCOPY_BUFFER_GB.",
```

Comment:
>AZCOPY_BUFFER_GB [](start = 77, length = 16)

minor: get the string for AZCOPY_BUFFER_GB by referring to the enum, so that in case we change it this place gets the update